R/mean.R

```diff
@@ -259,3 +259,54 @@ report.model_mean <- function(object, ...) {
 model_sum.model_mean <- function(x) {
   paste0("MEAN") # , ", intToUtf8(0x3BC), "=", format(x$par$estimate))
 }
+
+#' Refit a MEAN model
+#'
+#' Applies a fitted average method model to a new dataset.
+#'
+#' @inheritParams refit.ARIMA
+#' @param reestimate If `TRUE`, the mean for the fitted model will be re-estimated 
+#' to suit the new data. 
+#' 
+#' @examples
+#' lung_deaths_male <- as_tsibble(mdeaths)
+#' lung_deaths_female <- as_tsibble(fdeaths)
+#'
+#' fit <- lung_deaths_male %>%
+#'   model(MEAN(value))
+#'
+#' report(fit)
+#'
+#' fit %>%
+#'   refit(lung_deaths_female) %>%
+#'   report()
+#' @export
+refit.model_mean <- function(object, new_data, specials = NULL, reestimate = FALSE, ...) {
+  # Update data for re-evaluation
+  # update specials
+  specials$window <- if(is.na(object$window)) NULL else object$window 
+
+  if (reestimate) {
+    return(train_mean(new_data, specials, ...))
+  }
+  
+  y <- unclass(new_data)[[measured_vars(new_data)]]
+  
+  if (all(is.na(y))) {
+    abort("All new observations are missing, model cannot be applied.")
+  }
+
+  if (!is_null(specials$window)) warn("Rolling means are specified. The fixed mean is used to recalculate the fitted values!")
```

Comment:
This warning reads more like a message, informing the user what is happening.
A warning should explain why something is wrong, and what has been done about it.

Perhaps:
> A rolling mean model cannot be refitted, the most recent mean from the fitted model will be used as a fixed estimate of the mean.